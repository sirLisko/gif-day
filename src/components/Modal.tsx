import React from "react";
import styled from "@emotion/styled/macro";
import ReactModal from "react-modal2";
import { Global, css } from "@emotion/react";

import theme from "styles/theme";

const StyledModal = css`
  .modal {
    position: relative;
    width: 90%;
    height: 95%;
    margin: 0;
    padding: 2.5rem;
    background-color: #fff;
    border-radius: 0.3rem;
    box-shadow: 0 0 2.5rem rgba(0, 0, 0, 0.5);
    max-width: 700px;
    top: 50%;
    left: 50%;
    text-align: left;
    transform: translate3d(-50%, -50%, 0);
    ${theme.mq.small} {
      height: auto;
      width: 80%;
    }
    & > div {
      overflow-y: auto;
      height: 100%;
    }
  }
  .modal__backdrop {
    background-color: rgba(0, 0, 0, 0.7);
    top: 0;
    position: fixed;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 10;
  }
`;

const StyledCloseButton = styled("button")`
  position: fixed;
  top: 1rem;
  right: 1rem;
  border: none;
  padding: 0;
  font-size: 1.5rem;
  background: none;
`;

ReactModal.getApplicationElement = () => document.getElementById("root");

interface Props {
  isModalOpen?: boolean;
  onClose: () => void;
  children: React.ReactElement;
}

const Modal = ({ isModalOpen, onClose, children }: Props) => {
  if (!isModalOpen) {
    return null;
  }
  return (
    <ReactModal
      backdropClassName="modal__backdrop"
      modalClassName="modal"
      onClose={onClose}
    >
      <Global styles={StyledModal} />
      <StyledCloseButton onClick={onClose}>Ã—</StyledCloseButton>
      <div>{children}</div>
    </ReactModal>
  );
};

export default Modal;
