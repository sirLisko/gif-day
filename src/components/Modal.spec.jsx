import React from 'react';
import { shallow } from 'enzyme';

import Modal from './Modal';

describe('Modal Component', () => {
  let wrapper;
  const props = {
    isModalOpen: true,
    onClose: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(
      <Modal {...props}>
        <div>foo</div>
      </Modal>,
    );
  });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should NOT render if isModalOpen is false', () => {
    wrapper.setProps({ isModalOpen: false });
    expect(wrapper).toMatchSnapshot();
  });
});
