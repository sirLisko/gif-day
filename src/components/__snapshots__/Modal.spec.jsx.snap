// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should NOT render if isModalOpen is false 1`] = `""`;

exports[`Modal Component should render properly 1`] = `
<ReactModal2
  backdropClassName="modal__backdrop"
  closeOnBackdropClick={true}
  closeOnEsc={true}
  modalClassName="modal"
  onClose={[MockFunction]}
>
  <ForwardRef(render)
    styles={
      Object {
        "map": undefined,
        "name": "44yqo7",
        "next": undefined,
        "styles": "
  .modal {
    position: relative;
    width: 90%;
    height: 95%;
    margin: 0;
    padding: 2.5rem;
    background-color: #fff;
    border-radius: 0.3rem;
    box-shadow: 0 0 2.5rem rgba(0, 0, 0, 0.5);
    max-width: 700px;
    top: 50%;
    left: 50%;
    text-align: left;
    transform: translate3d(-50%, -50%, 0);
    @media (min-width: 620px) {
      height: auto;
      width: 80%;
    }
    & > div {
      overflow-y: auto;
      height: 100%;
    }
  }
  .modal__backdrop {
    background-color: rgba(0, 0, 0, 0.7);
    top: 0;
    position: fixed;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 10;
  }
",
      }
    }
  />
  <StyledCloseButton
    onClick={[MockFunction]}
  >
    Ã—
  </StyledCloseButton>
  <div>
    <div>
      foo
    </div>
  </div>
</ReactModal2>
`;
